cmd/kops/create_cluster.go

```diff
@@ -436,6 +424,33 @@ func RunCreateCluster(f *util.Factory, out io.Writer, c *CreateClusterOptions) e
 		}
 	}
 
+	cloud, err := cloudup.BuildCloud(cluster)
+	if err != nil {
+		return err
+	}
+
+	if c.VPCID != "" {
+		cluster.Spec.NetworkID = c.VPCID
+	}
+
+	if c.NetworkCIDR != "" {
+		cluster.Spec.NetworkCIDR = c.NetworkCIDR
+	}
+
+	if cluster.SharedVPC() && cluster.Spec.NetworkCIDR == "" {
+		vpcInfo, err := cloud.FindVPCInfo(cluster.Spec.NetworkID)
```

Comment:
You're absolutely right - it didn't belong in the CLI layer.  I moved it into PerformAssignments, where it really belongs... lots of logic in there that can use VPCInfo